### DISCUSSION TITLE
Improve documentation: transactions

### DISCUSSION THREAD

**Discussion by andriyndev** [serverpod/serverpod#878] [Ideas]
From the current documentation it is unclear how to use the transactions, i.e. how to call queries from inside `session.db.transaction((transaction) async { ... })`. From what I understand, we should pass `transaction` to the query function as a named argument, for example:
```
var result = await session.db.transaction((transaction) async {
  await TableObject1.insert(session, tableObj1, transaction: transaction);
  await TableObject2.insert(session, tableObj2, transaction: transaction);
  return true;
});
```
but the documentation doesn't state that we need to pass `transaction` to the query calls, so a developer may not know that they need to do it, and just call
```
var result = await session.db.transaction((transaction) async {
  await TableObject1.insert(session, tableObj1);
  await TableObject2.insert(session, tableObj2);
  return true;
});
```
that, from what I understand, isn't a transaction.
This should be clarified in the documentation.

**Comment by vlidholt** [comment]
I added an issue for this so that we won't forget! Thank you, will dig into this and check that the docs are correct. :)
